lib/supported-languages.nix

```diff
@@ -0,0 +1,279 @@
+{pkgs, evalPackages, ghc}:
+evalPackages.writeTextFile {
+  name = "supported-languages";
```

Comment:
Don't these depend on the compiler version?

Reply:
Logic is added now

Reply:
Oh and there is a test now to make sure it is correct by comparing the results to the `ghc --supported-languages` output.